virtcontainers/fc.go

```diff
@@ -737,30 +739,33 @@ func (fc *firecracker) resumeSandbox() error {
 	return nil
 }
 
-func (fc *firecracker) fcAddVsock(vs kataVSOCK) error {
-	span, _ := fc.trace("fcAddVsock")
+func (fc *firecracker) fcAddVsock(hvs types.HybridVSock) error {
+	span, _ := fc.trace("fcAddHybridVsock")
```

Comment:
Just thinking out loud here, do we want this change to be backward compatible, ie use the correct vsock type based on the version of firecracker? At the very least do we check beforehand that the hypervisor supports virtio-vsock and throw a friendly error message asking the user to upgrade version of firecracker?

Reply:
Good question, IMO it'd be a pain maintaining backward compatibility with all the versions of a project like FC that may introduce changes like this in each release. WDYT?

cc @sboeuf @egernst

Reply:
No I don't think so. This new implementation of vsock was waited for a long time from the FC folks, and now they have it, they won't move back. For that reason, we shouldn't bother with backward compatibility here.

Reply:
@sboeuf I was just referring to this in the context of having an updated runtime, but an older version of firecracker on a system. Maybe this is something that may happen on a dev machine, the runtime could  just check that we are running an older version of firecracker and throw a friendly error message.

Reply:
@amshinde that sounds like a great proposal :+1: